save-backend/src/main/kotlin/org/cqfn/save/backend/repository/TimestampBasedFileSystemRepository.kt

```diff
@@ -125,7 +126,7 @@ class TimestampBasedFileSystemRepository(configProperties: ConfigProperties) {
                     .collect(Collectors.summingLong { it })
                     .map {
                         logger.info("Saved $it bytes into $this")
-                        val relativePath = ("/$owner/$name")
+                        val relativePath = if (isOrganization) ("/$imageName/$name") else ("/users/$imageName/$name")
```

Comment:
You are duplicating logic of storage path construction in several places in this class. It should be extracted into a method

Reply:
In this place, a relative path to the file itself is collected. And in another case, the full path to the directory itself

Reply:
Yes, but the directory name `users` is duplicated. You can extract relative path creation logic into a method, which accepts AvatarType, and for absolute paths call `rootDir.resolve(methodForRelativePath(type))`

Reply:
Fixed